datafusion-examples/examples/relation_planner/table_sample.rs

```diff
@@ -0,0 +1,969 @@
+struct Bound(f64);
+
+impl PartialEq for Bound {
+    fn eq(&self, other: &Self) -> bool {
+        (self.0 - other.0).abs() < f64::EPSILON
+    }
+}
+
+impl Eq for Bound {}
+
+impl Hash for Bound {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Hash the bits of the f64
+        self.0.to_bits().hash(state);
+    }
+}
+
+impl From<f64> for Bound {
+    fn from(value: f64) -> Self {
+        Self(value)
+    }
+}
+impl From<Bound> for f64 {
+    fn from(value: Bound) -> Self {
+        value.0
+    }
+}
+
+impl AsRef<f64> for Bound {
+    fn as_ref(&self) -> &f64 {
+        &self.0
+    }
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, PartialOrd)]
+struct TableSamplePlanNode {
+    inner_plan: LogicalPlan,
+
+    lower_bound: Bound,
+    upper_bound: Bound,
+    with_replacement: bool,
+    seed: u64,
+}
+
+impl TableSamplePlanNode {
+    pub fn new(
+        input: LogicalPlan,
+        fraction: f64,
+        with_replacement: Option<bool>,
+        seed: Option<u64>,
+    ) -> Self {
+        TableSamplePlanNode {
+            inner_plan: input,
+            lower_bound: Bound::from(0.0),
+            upper_bound: Bound::from(fraction),
+            with_replacement: with_replacement.unwrap_or(false),
+            seed: seed.unwrap_or_else(rand::random),
+        }
+    }
+
+    pub fn into_plan(self) -> LogicalPlan {
+        LogicalPlan::Extension(Extension {
+            node: Arc::new(self),
+        })
+    }
+}
+
+impl UserDefinedLogicalNodeCore for TableSamplePlanNode {
+    fn name(&self) -> &str {
+        "TableSample"
+    }
+
+    fn inputs(&self) -> Vec<&LogicalPlan> {
+        vec![&self.inner_plan]
+    }
+
+    fn schema(&self) -> &DFSchemaRef {
+        self.inner_plan.schema()
+    }
+
+    fn expressions(&self) -> Vec<Expr> {
+        vec![]
+    }
+
+    fn fmt_for_explain(&self, f: &mut Formatter) -> fmt::Result {
+        f.write_fmt(format_args!(
+            "Sample: {:?} {:?} {:?}",
+            self.lower_bound, self.upper_bound, self.seed
+        ))
+    }
+
+    fn with_exprs_and_inputs(
+        &self,
+        _exprs: Vec<Expr>,
+        inputs: Vec<LogicalPlan>,
+    ) -> Result<Self> {
+        let input = inputs
+            .first()
+            .ok_or(DataFusionError::Plan("Should have input".into()))?;
+        Ok(Self {
+            inner_plan: input.clone(),
+            lower_bound: self.lower_bound,
+            upper_bound: self.upper_bound,
+            with_replacement: self.with_replacement,
+            seed: self.seed,
+        })
+    }
+}
+
+/// Execution planner with `SampleExec` for `TableSamplePlanNode`
+struct TableSampleExtensionPlanner {}
+
+impl TableSampleExtensionPlanner {
+    fn build_execution_plan(
+        &self,
+        specific_node: &TableSamplePlanNode,
+        physical_input: &Arc<dyn ExecutionPlan>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        Ok(Arc::new(SampleExec {
+            input: Arc::clone(physical_input),
+            lower_bound: 0.0,
```

Comment:
Why 0.0 ? Why not `specific_node.lower_bound.into()` ?
Maybe add the reason as a comment.